tests/mpi/tria_copy_triangulation_02.cc

```diff
@@ -0,0 +1,130 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// Same as tria_copy_triangulation_01, but for locally refined meshes.
+
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/grid/grid_generator.h>
+
+#include "../tests.h"
+
+
+template <int dim>
+void
+test()
+{
+  unsigned int myid = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+
+  parallel::distributed::Triangulation<dim> tria1(MPI_COMM_WORLD);
+  {
+    GridGenerator::hyper_cube(tria1);
```

Comment:
Could you use here a geometry with a manifold attached to? Like hyper_ball? And maybe also set the boundary ids?

Reply:
I used `hyper_shell` with the `colorize` option and added a few assertions. Looks good to me now!